tests/functional/storageclass/test_create_storageclass_with_same_name.py

```
@@ -60,16 +60,19 @@ def create_storageclass(sc_name, expect_fail=False):
+            f"SC creation with name '{sc_name}': expected_fail='{expect_fail}', actual_fail='False'"
+        )
         assert not expect_fail, "SC creation with same name passed. Expected to fail !"
-        log.info(f"Storage class: {SC_OBJ.name} created successfully !")
-        log.debug(sc_data)
+        logger.info(f"Storage class: {SC_OBJ.name} created successfully")
+        logger.debug(f"Storage class data: {sc_data}")
 
     except CommandFailed as ecf:
+        if not expect_fail:
+            raise
         assert "AlreadyExists" in str(ecf)
```

Ruff (0.15.15)

[warning] Found assertion on exception `ecf` in `except` block, use `pytest.raises()` instead

(PT017)